src/apps/core/app.js

```diff
@@ -89,13 +90,35 @@ define(function(require) {
 			self.displayFavicon();
 			self.loadSVG();
 
+			if (crossSiteMessaging !== undefined) {
+				self.bindCrossSiteMessagingHandler(crossSiteMessaging);
+			}
+
 			container.append(mainTemplate);
 
 			monster.waterfall([
 				_.bind(self.loadIncludes, self)
 			], _.bind(self.loadAuth, self));
 		},
 
+		bindCrossSiteMessagingHandler: function(crossSiteMessaging) {
+			const { origin, topics } = crossSiteMessaging;
+
+			window.addEventListener('message', handler);
+
+			function handler(event) {
+				if (event.origin !== origin || !topics.includes(event.data)) {
```

Comment:
Only a single origin will be allowed? Is it possible or necessary to allow more than one origin?

Also, another thought, what if we leave to each application to check which topics are allowed for cross site messaging?